src/Rings/mpoly-ideals.jl

```diff
@@ -1191,17 +1197,27 @@ function minimal_primes(
 
   # This will in many cases lead to an easy simplification of the problem
   if factor_generators
-    J = typeof(I)[ideal(R, elem_type(R)[])]
+    J = [ideal(R, gens(I))] # A copy of I as initialization
     for g in gens(I)
       K = typeof(I)[]
       is_zero(g) && continue
       for (b, k) in factor(g)
+        # Split the already collected components with b
         for j in J
           push!(K, j + ideal(R, b))
         end
       end
       J = K
     end
+
+    unique_comp = typeof(I)[]
+    for q in J
+      is_one(q) && continue
+      q in unique_comp && continue
```

Comment:
This compares ideals, won't that trigger additional GB computations and thus be slow in general?? Maybe you just want to remove "obviously" identical ideals (with generators lists being the same, up to reordering?)

Reply:
@fingolfin : is_one(q) triggers a GB computation, but (at least hopefully) the ideal membership test in 1240 will use that GB for the comparison.

I rather wonder whether the factorization in Oscar is so much superior that it does not make sense to do a `facstd` of J in Singular and then go on with the components from there.